libs/ngworker/lumberjack/src/lib/logging/lumberjack-log.builder.ts

```diff
@@ -0,0 +1,48 @@
+import { LumberjackLogLevel } from '../logs/lumberjack-log-level';
+import { LumberjackLogPayload } from '../logs/lumberjack-log-payload';
+import { LumberjackLog } from '../logs/lumberjack.log';
+import { LumberjackTimeService } from '../time/lumberjack-time.service';
+
+/**
+ * Util type that helps to identify if the `withPayload` method was used.
+ */
+type StaticPayload = 'static-payload' & LumberjackLogPayload;
```

Comment:
Can we use this name and string representation instead? Also rename the type elsewhere in this file.
```suggestion
/**
 * The generic parameter of `LumberjackLogBuilder` is evaluated to this type
 * when the `withPayload` method is used.
 */
type InternalWithStaticPayload = '__LUMBERJACK_INTERNAL_WITH_STATIC_PAYLOAD__' & LumberjackLogPayload;
```